core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -310,16 +311,14 @@ object FileUtils {
     return null
   }
 
-  /**
-   * Returns the file name from the url or src. In url it gets the file name from the last '/' and
-   * if it contains '.'. If the url is null then it'll get the file name from the last '/'.
-   * If the url and src doesn't exist it returns the empty string.
-   */
-  fun getDecodedFileName(url: String?, src: String?): String =
-    url?.substringAfterLast("/", "")
-      ?.takeIf { it.contains(".") }
-      ?: src?.substringAfterLast("/", "")
-        ?.substringAfterLast("%3A") ?: ""
+  fun getDecodedFileName(url: String?, src: String?): String {
+    var fileName = ""
+    val decodedFileName = URLUtil.guessFileName(url ?: src, null, null)
+    if (decodedFileName != "downloadfile.bin") {
```

Comment:
Why!?! What is ""downloadfile.bin"?!?

Reply:
@kelson42 ,
it is default value from that method, if couldn't get filename from provided url or src.

Reply:
This is a fundamental change of the behaviour which not even documented! Actually there is not even a PR description. That's the first problem.

Then I disagree, if there is not Filename then the method should return `undefined` (or worse case empty string), all theses scenario should be part of the automated tests as well.

Reply:
@kelson42 , I have changed it. if the there is no filename or if ```URLUtils.guessFileName()``` returns the default filename (downloadfile.bin) then this method returns the null value , which is handled in ```downloadFileFromUrl ``` method. i have added unit test for this condition.